hack/filter-crd/main.go

```diff
@@ -52,13 +55,37 @@ func main() {
 
 		checkChain(d, []string{})
 
+		if checkValidationLocation {
+			spec, ok := d["spec"].(map[interface{}]interface{})
+			if !ok {
+				log.Fatal("Cannot read spec of CRD")
+			}
+			versions, ok := spec["versions"].([]interface{})
+			if !ok {
+				log.Fatal("Cannot read versions of CRD")
+			}
+			if len(versions) == 0 {
+				log.Fatal("CRD versions length is 0")
+			}
+			versionInfo, ok := versions[0].(map[interface{}]interface{})
```

Comment:
How do we know that `version[0]` is always `v1alpha2`? Seems like this works so long as the correct `removeElementForValue` are specified to remove the others, but it feels slightly fragile because if we add a new version (v1alpha4), we will now remove v1alpha3 and then promote *something* to the top-level, causing the CRD to be invalid 😬

Might be okay to just leave, but probably worth a comment at the very least